src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -250,6 +285,277 @@ impl<'a, B: CryptoOps> Policy<'a, B> {
+                    Criticality::Agnostic,
+                    Some(ee::subject_alternative_name),
+                ),
+                // 5280 4.2.1.9: Basic Constraints
+                ExtensionPolicy::maybe_present(
+                    BASIC_CONSTRAINTS_OID,
+                    Criticality::Agnostic,
+                    Some(ee::basic_constraints),
+                ),
+                // 5280 4.2.1.10: Name Constraints
+                ExtensionPolicy::not_present(NAME_CONSTRAINTS_OID),
+                // CA/B 7.1.2.7.10 Subscriber Certificate Extended Key Usage
+                ExtensionPolicy::present(
+                    EXTENDED_KEY_USAGE_OID,
+                    Criticality::NonCritical,
+                    Some(ee::extended_key_usage),
+                ),
+            ]),
+        }
+    }
+
+    fn permits_basic(&self, cert: &Certificate<'_>) -> Result<(), PolicyError> {
+        let extensions = cert.extensions()?;
+
+        // CA/B 7.1.1:
+        // Certificates MUST be of type X.509 v3.
+        if cert.tbs_cert.version != 2 {
+            return Err("certificate must be an X509v3 certificate".into());
+        }
+
+        // 5280 4.1.1.2 / 4.1.2.3: signatureAlgorithm / TBS Certificate Signature
+        // The top-level signatureAlgorithm and TBSCert signature algorithm
+        // MUST match.
+        if cert.signature_alg != cert.tbs_cert.signature_alg {
+            return Err("mismatch between signatureAlgorithm and SPKI algorithm".into());
+        }
+
+        // 5280 4.1.2.2: Serial Number
+        let serial_bytes = cert.tbs_cert.serial.as_bytes();
+        if serial_bytes.len() == 1 && serial_bytes[0] == 0 {
+            // The serial number MUST be a positive integer.
+            return Err("certificate serial number must not be 0".into());
+        } else if !(1..=21).contains(&serial_bytes.len()) {
+            // Conforming CAs MUST NOT use serial numbers longer than 20 octets.
+            // NOTE: In practice, this requires us to check for an encoding of
+            // 21 octets, since some CAs generate 20 bytes of randomness and
+            // then forget to check whether that number would be negative, resulting
+            // in a 21-byte encoding.
+            return Err("certificate must have a serial between 1 and 20 octets".into());
+        }
+
+        // 5280 4.1.2.4: Issuer
+        // The issuer MUST be a non-empty distinguished name.
+        if cert.issuer().is_empty() {
+            return Err("certificate must have a non-empty Issuer".into());
+        }
+
+        // 5280 4.1.2.5: Validity
+        // Validity dates before 2050 MUST be encoded as UTCTime;
+        // dates in or after 2050 MUST be encoded as GeneralizedTime.
+        // TODO: The existing `tbs_cert.validity` types don't expose this
+        // underlying detail. This check has no practical effect on the
+        // correctness of the certificate, so it's pretty low priority.
```

Comment:
Not sure I understand this, they're Time instances so you can see what tlv type they have.

Reply:
[90162e4](https://github.com/pyca/cryptography/pull/8873/commits/90162e436fd38dad887f597d5cdf31bc0489c727)

Reply:
[adc7333](https://github.com/pyca/cryptography/pull/8873/commits/adc7333ec9e0022d819c75e12fe25af70f722fce)